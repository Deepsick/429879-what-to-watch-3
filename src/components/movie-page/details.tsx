import * as React from 'react';

interface Props {
  director: string;
  starring: string[];
  runTime: number;
  genre: string;
  released: number;
}

const Details: React.FunctionComponent<Props> = ({
  director,
  starring,
  runTime,
  genre,
  released
}: Props) => (
  <div className="movie-card__text movie-card__row">
    <div className="movie-card__text-col">
      <p className="movie-card__details-item">
        <strong className="movie-card__details-name">Director</strong>
        <span className="movie-card__details-value">{director}</span>
      </p>
      <p className="movie-card__details-item">
        <strong className="movie-card__details-name">Starring</strong>
        <span className="movie-card__details-value">
          {starring.map((actor) => actor).join(`,\n`)}
        </span>
      </p>
    </div>

    <div className="movie-card__text-col">
      <p className="movie-card__details-item">
        <strong className="movie-card__details-name">Run Time</strong>
        <span className="movie-card__details-value">{runTime}</span>
      </p>
      <p className="movie-card__details-item">
        <strong className="movie-card__details-name">Genre</strong>
        <span className="movie-card__details-value">{genre}</span>
      </p>
      <p className="movie-card__details-item">
        <strong className="movie-card__details-name">Released</strong>
        <span className="movie-card__details-value">{released}</span>
      </p>
    </div>
  </div>
);

export default React.memo(Details);
