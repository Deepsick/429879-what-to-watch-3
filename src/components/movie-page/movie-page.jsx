import React, {Fragment, memo} from 'react';
import PropTypes from 'prop-types';
import Tabs, {TabName} from '../tabs/tabs.jsx';
import MoreLikeThis from '../more-like-this/more-like-this.jsx';
import Footer from '../footer/footer.jsx';
import withActiveItem from '../../hocs/with-active-item/with-active-item.jsx';

const TabsWrapped = withActiveItem(Tabs);

const handlePlayButtonClick = (id, callback) => () => {
  callback(id);
};

const MoviePage = ({movie, similarMovies, onPlayButtonClick}) => {
  const {
    name,
    genre,
    year,
    poster,
    preview,
    director,
    bgColor,
    starring,
    description,
    id,
    rating,
    scoresCount,
  } = movie;
  return (
    <Fragment>
      <section className="movie-card movie-card--full" style={{backgroundColor: bgColor}}>
        <div className="movie-card__hero">
          <div className="movie-card__bg">
            <img src={poster} alt={name} />
          </div>

          <h1 className="visually-hidden">WTW</h1>

          <header className="page-header movie-card__head">
            <div className="logo">
              <a href="main.html" className="logo__link">
                <span className="logo__letter logo__letter--1">W</span>
                <span className="logo__letter logo__letter--2">T</span>
                <span className="logo__letter logo__letter--3">W</span>
              </a>
            </div>

            <div className="user-block">
              <div className="user-block__avatar">
                <img src="img/avatar.jpg" alt="User avatar" width="63" height="63" />
              </div>
            </div>
          </header>

          <div className="movie-card__wrap">
            <div className="movie-card__desc">
              <h2 className="movie-card__title">{name}</h2>
              <p className="movie-card__meta">
                <span className="movie-card__genre">{genre}</span>
                <span className="movie-card__year">{year}</span>
              </p>

              <div className="movie-card__buttons">
                <button
                  className="btn btn--play movie-card__button"
                  type="button"
                  onClick={handlePlayButtonClick(id, onPlayButtonClick)}
                >
                  <svg viewBox="0 0 19 19" width="19" height="19">
                    <use xlinkHref="#play-s"></use>
                  </svg>
                  <span>Play</span>
                </button>
                <button className="btn btn--list movie-card__button" type="button">
                  <svg viewBox="0 0 18 14" width="18" height="14">
                    <use xlinkHref="#in-list"></use>
                  </svg>
                  <span>My list</span>
                </button>
                <a href="add-review.html" className="btn movie-card__button">Add review</a>
              </div>
            </div>
          </div>
        </div>

        <div className="movie-card__wrap movie-card__translate-top">
          <div className="movie-card__info">
            <div className="movie-card__poster movie-card__poster--big">
              <img src={preview} alt={name} width="218" height="327" />
            </div>

            <div className="movie-card__desc">
              <TabsWrapped activeTab={TabName.OVERVIEW} />

              <div className="movie-rating">
                <div className="movie-rating__score">{rating}</div>
                <p className="movie-rating__meta">
                  <span className="movie-rating__level">Very good</span>
                  <span className="movie-rating__count">{scoresCount} ratings</span>
                </p>
              </div>

              <div className="movie-card__text">
                <p>{description}</p>
                <p className="movie-card__director"><strong>Director: {director}</strong></p>

                <p className="movie-card__starring"><strong>Starring: {starring.join(`, `)} and other</strong></p>
              </div>
            </div>
          </div>
        </div>
      </section>
      <div className="page-content">
        <MoreLikeThis movies={similarMovies} />
        <Footer />
      </div>
    </Fragment>
  );
};

MoviePage.propTypes = {
  movie: PropTypes.exact({
    name: PropTypes.string.isRequired,
    poster: PropTypes.string.isRequired,
    preview: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    rating: PropTypes.number.isRequired,
    scoresCount: PropTypes.number.isRequired,
    isFavorite: PropTypes.bool.isRequired,
    trailer: PropTypes.string.isRequired,
    video: PropTypes.string.isRequired,
    duration: PropTypes.number.isRequired,
    year: PropTypes.number.isRequired,
    cover: PropTypes.string.isRequired,
    bgColor: PropTypes.string.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.arrayOf(PropTypes.string).isRequired,
    description: PropTypes.string.isRequired,
  }).isRequired,
  similarMovies: PropTypes.arrayOf(PropTypes.exact({
    name: PropTypes.string.isRequired,
    poster: PropTypes.string.isRequired,
    preview: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    rating: PropTypes.number.isRequired,
    scoresCount: PropTypes.number.isRequired,
    isFavorite: PropTypes.bool.isRequired,
    trailer: PropTypes.string.isRequired,
    video: PropTypes.string.isRequired,
    duration: PropTypes.number.isRequired,
    year: PropTypes.number.isRequired,
    cover: PropTypes.string.isRequired,
    bgColor: PropTypes.string.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.arrayOf(PropTypes.string).isRequired,
    description: PropTypes.string.isRequired,
  })).isRequired,
  onPlayButtonClick: PropTypes.func.isRequired,
};

export default memo(MoviePage);
