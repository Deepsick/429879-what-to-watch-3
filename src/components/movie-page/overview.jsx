import React, {Fragment, memo} from 'react';

const RatingLimits = {
  BAD: 0,
  NORMAL: 3,
  GOOD: 5,
  VERY_GOOD: 8,
  AWESOME: 10,
};

const RatingLevel = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
};

const getRatingLevel = (rating) => {
  const roundedRating = Math.floor(rating);

  if (roundedRating < RatingLimits.NORMAL) {
    return RatingLevel.BAD;
  }

  if (roundedRating < RatingLimits.GOOD) {
    return RatingLevel.NORMAL;
  }

  if (roundedRating < RatingLimits.VERY_GOOD) {
    return RatingLevel.GOOD;
  }

  return RatingLevel.AWESOME;
};

const Overview = ({rating, scoresCount, director, description, starring}) => (
  <Fragment>
    <div className="movie-rating">
      <div className="movie-rating__score">{rating}</div>
      <p className="movie-rating__meta">
        <span className="movie-rating__level">{getRatingLevel(rating)}</span>
        <span className="movie-rating__count">{scoresCount} ratings</span>
      </p>
    </div>

    <div className="movie-card__text">
      <p>{description}</p>
      <p className="movie-card__director"><strong>Director: {director}</strong></p>

      <p className="movie-card__starring"><strong>Starring: {starring.join(`, `)} and other</strong></p>
    </div>
  </Fragment>
);

export default memo(Overview);
