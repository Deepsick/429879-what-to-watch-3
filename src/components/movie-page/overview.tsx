import * as React from 'react';

interface Props {
  rating: number;
  scoresCount: number;
  director: string;
  description: string;
  starring: string[];
}

const RatingLimits = {
  BAD: 0,
  NORMAL: 3,
  GOOD: 5,
  VERY_GOOD: 8,
  AWESOME: 10,
};

const RatingLevel = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
};

const getRatingLevel = (rating) => {
  const roundedRating = Math.floor(rating);

  if (roundedRating < RatingLimits.NORMAL) {
    return RatingLevel.BAD;
  }

  if (roundedRating < RatingLimits.GOOD) {
    return RatingLevel.NORMAL;
  }

  if (roundedRating < RatingLimits.VERY_GOOD) {
    return RatingLevel.GOOD;
  }

  return RatingLevel.AWESOME;
};

const Overview: React.FunctionComponent<Props> = ({
  rating,
  scoresCount,
  director,
  description,
  starring
}: Props) => (
  <React.Fragment>
    <div className="movie-rating">
      <div className="movie-rating__score">{rating}</div>
      <p className="movie-rating__meta">
        <span className="movie-rating__level">{getRatingLevel(rating)}</span>
        <span className="movie-rating__count">{scoresCount} ratings</span>
      </p>
    </div>

    <div className="movie-card__text">
      <p>{description}</p>
      <p className="movie-card__director"><strong>Director: {director}</strong></p>

      <p className="movie-card__starring"><strong>Starring: {starring.join(`, `)} and other</strong></p>
    </div>
  </React.Fragment>
);

export default React.memo(Overview);
