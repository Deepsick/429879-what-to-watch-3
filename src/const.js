export const ALL_GENRES = `All genres`;
export const MOVIES_COUNT = 8;
export const MAX_GENRES_COUNT = 9;
export const START_INDEX = 0;
export const SECONDS_IN_MINUTE = 60;
export const MAX_PERCENTAGE = 100;
export const MAX_SIMILAR_MOVIES_COUNT = 4;
export const Api = {
  BASE_URL: `https://htmlacademy-react-3.appspot.com/wtw`,
  TIMEOUT: 5000,
};

export const Auth = {
  NO_AUTH: `NO_AUTH`,
  AUTH: `AUTH`,
};

export const ReducerName = {
  DATA: `data`,
  STATE: `state`,
  USER: `user`,
};

export const Path = {
  FILMS: `films`,
  LOGIN: `login`,
};

export const Status = {
  OK: 200,
  UNAUTHORIZED: 401,
  BAD_REQUEST: 400,
};

