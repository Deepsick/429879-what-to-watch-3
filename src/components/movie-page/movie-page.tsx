import * as React from 'react';
import {connect} from 'react-redux';

import Tabs from '../tabs/tabs';
import MoreLikeThis from '../more-like-this/more-like-this';
import Details from './details';
import Overview from './overview';
import Reviews from './reviews';
import Footer from '../footer/footer';
import Logo from '../logo/logo';
import Spinner from '../spinner/spinner';
import Avatar from '../avatar/avatar';
import CardButtons from '../card-buttons/card-buttons';
import {Operation} from '../../reducer/data/data';
import {getMovies, getComments} from '../../reducer/data/selectors';
import {getAuthStatus, getAvatar} from '../../reducer/user/selectors';
import {START_INDEX, MAX_SIMILAR_MOVIES_COUNT, TabName} from '../../const';
import {Movie, CommentGet} from '../../types';

interface Props {
  movies: Movie[];
  id: number;
  active: string;
  isAuth: string;
  avatar: string;
  comments: CommentGet[];
  loadComments: (id: number) => void;
  postFavorite: (id: number, status: boolean) => void;
  setActiveItem: (item: string) => void;
}

class MoviePage extends React.PureComponent<Props, {}> {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const {id, loadComments} = this.props;
    loadComments(id);
  }

  _getFilmById() {
    const {movies, id} = this.props;
    return movies.find((movie) => movie.id === +id);
  }

  _getSimilarMovies(genre) {
    const {movies, id} = this.props;
    return movies
      .filter((movie) => movie.genre === genre && movie.id !== +id)
      .slice(START_INDEX, MAX_SIMILAR_MOVIES_COUNT);
  }

  render() {
    const {postFavorite, comments, active, setActiveItem, isAuth, avatar, movies} = this.props;

    if (movies.length === 0) {
      return <Spinner />;
    }

    const movie = this._getFilmById();
    const {
      name,
      genre,
      year,
      poster,
      director,
      duration,
      bgColor,
      starring,
      description,
      id,
      isFavorite,
      rating,
      scoresCount,
      cover,
    } = movie;
    const similarMovies = this._getSimilarMovies(genre);

    return (
      <React.Fragment>
        <section className="movie-card movie-card--full" style={{backgroundColor: bgColor}}>
          <div className="movie-card__hero">
            <div className="movie-card__bg">
              <img src={cover} alt={name} />
            </div>

            <h1 className="visually-hidden">WTW</h1>
            <header className="page-header movie-card__head">
              <Logo />
              <Avatar isAuth={isAuth} avatar={avatar} />
            </header>

            <div className="movie-card__wrap">
              <div className="movie-card__desc">
                <h2 className="movie-card__title">{name}</h2>
                <p className="movie-card__meta">
                  <span className="movie-card__genre">{genre}</span>
                  <span className="movie-card__year">{year}</span>
                </p>
                <CardButtons
                  isAddReview={true}
                  onClick={postFavorite}
                  isFavorite={isFavorite}
                  id={id} />
              </div>
            </div>
          </div>

          <div className="movie-card__wrap movie-card__translate-top">
            <div className="movie-card__info">
              <div className="movie-card__poster movie-card__poster--big">
                <img src={poster} alt={name} width="218" height="327" />
              </div>

              <div className="movie-card__desc">
                <Tabs setActiveItem={setActiveItem} active={active} />

                {active === TabName.OVERVIEW &&
                  <Overview
                    rating={rating}
                    scoresCount={scoresCount}
                    director={director}
                    description={description}
                    starring={starring}
                  />
                }

                {active === TabName.DETAILS &&
                  <Details
                    director={director}
                    starring={starring}
                    genre={genre}
                    released={year}
                    runTime={duration}
                  />
                }

                {active === TabName.REVIEWS &&
                  <Reviews comments={comments}/>
                }
              </div>
            </div>
          </div>
        </section>
        <div className="page-content">
          <MoreLikeThis movies={similarMovies} />
          <Footer />
        </div>
      </React.Fragment>
    );
  }
}

const mapStateToProps = (state) => ({
  movies: getMovies(state),
  comments: getComments(state),
  isAuth: getAuthStatus(state),
  avatar: getAvatar(state),
});

const mapDispatchToProps = (dispatch) => ({
  loadComments(id) {
    dispatch(Operation.loadComments(id));
  },
  postFavorite(id, status) {
    dispatch(Operation.postFavorite(id, status));
  },
});

export {MoviePage};
export default connect(mapStateToProps, mapDispatchToProps)(MoviePage);

