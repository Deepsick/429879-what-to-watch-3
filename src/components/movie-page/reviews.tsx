import * as React from 'react';

import {START_INDEX, REVIEW_COLS_COUNT} from '../../const';
import {CommentGet} from '../../types';

interface Props {
  comments: CommentGet[];
}

const formatDate = (date) => {
  const jsDate = new Date(date);
  const month = jsDate.toLocaleString(`default`, {month: `long`});
  const year = jsDate.getFullYear();
  const day = jsDate.getDate();

  return `${month} ${day}, ${year}`;
};

const getCommentMarkup = (comment) => {
  const {id, rating, comment: text, user, date} = comment;
  return (
    <div className="review" key={id}>
      <blockquote className="review__quote">
        <p className="review__text">{text}</p>

        <footer className="review__details">
          <cite className="review__author">{user.name}</cite>
          <time className="review__date" dateTime={date}>{formatDate(date)}</time>
        </footer>
      </blockquote>

      <div className="review__rating">{rating}</div>
    </div>
  );
};

const getCommentsMarkup = (comments) => {
  const reviewsInRow = Math.ceil(comments.length / REVIEW_COLS_COUNT);
  let start = START_INDEX;
  let finish = reviewsInRow;

  return new Array(REVIEW_COLS_COUNT).fill(`*`).map((col, index) => {
    const row = (
      <div key={index} className="movie-card__reviews-col">
        {comments.slice(start, finish).map((comment) => {
          return getCommentMarkup(comment);
        })}
      </div>
    );
    start += reviewsInRow;
    finish += reviewsInRow;
    return row;
  });
};

const Reviews: React.FunctionComponent<Props> = ({comments}: Props) => (
  <div className="movie-card__reviews movie-card__row">
    {getCommentsMarkup(comments)}
  </div>
);

export default React.memo(Reviews);
