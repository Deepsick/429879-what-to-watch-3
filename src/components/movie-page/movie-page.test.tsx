import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {BrowserRouter} from 'react-router-dom';

import {MoviePage} from './movie-page';
import {movies, id, mockFunction, mockString, Auth, comments} from '../../mocks/test-data';
import {ReducerName} from '../../const';

const store = configureStore([])({
  [ReducerName.DATA]: {
    movies,
    comments,
  },
  [ReducerName.USER]: {
    authorizationStatus: Auth.AUTH,
    authInfo: {
      avatar: `hello.jpg`,
    },
  },
});

it(`Should render MoviePage component correctly`, () => {
  const node = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <MoviePage
            movies={movies}
            id={id}
            active={mockString}
            isAuth={Auth.AUTH}
            avatar={mockString}
            comments={comments}
            loadComments={mockFunction}
            onFavoriteButtonClick={mockFunction}
            setActiveItem={mockFunction}
          />
        </BrowserRouter>
      </Provider>
  ).toJSON();

  expect(node).toMatchSnapshot();
});
