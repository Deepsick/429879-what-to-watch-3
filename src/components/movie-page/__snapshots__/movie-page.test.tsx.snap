// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MoviePage component correctly 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": "#ffffff",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Fantastic Beasts: The Crimes of Grindelwald"
          src="bg-the-grand-budapest-hotel.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <div
            className="user-block__avatar"
          >
            <a
              href="/mylist"
              onClick={[Function]}
            >
              <img
                alt="User avatar"
                height="63"
                src="https://htmlacademy-react-3.appspot.com/"
                width="63"
              />
            </a>
          </div>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Fantastic Beasts: The Crimes of Grindelwald
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              drama
            </span>
            <span
              className="movie-card__year"
            >
              2014
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/1"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="14"
                viewBox="0 0 18 14"
                width="18"
              >
                <use
                  xlinkHref="#in-list"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              className="btn movie-card__button"
              href="/films/1/review"
              onClick={[Function]}
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Fantastic Beasts: The Crimes of Grindelwald"
            height="327"
            src="fantastic-beasts-the-crimes-of-grindelwald.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item "
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item "
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item "
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseOut={[Function]}
        >
          <a
            href="/films/2"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <img
                alt="Bohemian Rhapsody"
                height="175"
                src=""
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
            >
              Bohemian Rhapsody
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseOut={[Function]}
        >
          <a
            href="/films/3"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <img
                alt="Macbeth"
                height="175"
                src=""
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/3"
              onClick={[Function]}
            >
              Macbeth
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseOut={[Function]}
        >
          <a
            href="/films/4"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <img
                alt="Aviator"
                height="175"
                src=""
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/4"
              onClick={[Function]}
            >
              Aviator
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseOut={[Function]}
        >
          <a
            href="/films/5"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <img
                alt="We need to talk about Kevin"
                height="175"
                src=""
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/5"
              onClick={[Function]}
            >
              We need to talk about Kevin
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
